engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceTest.java

```diff
@@ -222,4 +227,63 @@ public void shouldCompleteUndefinedTask() {
             tuple(BpmnElementType.TASK, ProcessInstanceIntent.ELEMENT_COMPLETED),
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
   }
+
+  @Test
+  public void shouldActivateNoneStartEventWhenEventTriggerIsAvailableInState() {
+    // given
+    final BpmnModelInstance modelInstance =
+        Bpmn.createExecutableProcess("process")
+            .startEvent("noneStart")
+            .endEvent()
+            .moveToProcess("process")
+            .startEvent("msgStart")
+            .message("msg")
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(modelInstance).deploy();
+
+    // when
+    ENGINE.writeRecords(
+        RecordToWrite.command()
+            .processInstanceCreation(
+                ProcessInstanceCreationIntent.CREATE,
+                new ProcessInstanceCreationRecord().setBpmnProcessId("process").setVersion(1)),
+        RecordToWrite.command()
+            .message(
+                MessageIntent.PUBLISH,
+                new MessageRecord().setName("msg").setCorrelationKey("").setTimeToLive(0)));
+
+    final var processInstanceKey =
+        RecordingExporter.processInstanceCreationRecords()
+            .withIntent(ProcessInstanceCreationIntent.CREATED)
+            .getFirst()
+            .getValue()
+            .getProcessInstanceKey();
```

Comment:
:x: Getting the first process instance creation record with a class rule is flaky.

If the test case is not executed first then the first record is written by a different test case. If I run the whole test class locally then the test case fails. 

We could use a unique process id (e.g. `BrokerClassRuleHelper.getBpmnProcessId()`) and use it to filter the records.